lib/tasks.js

```diff
@@ -368,10 +367,10 @@ module.exports = Emitter => {
 
 function resolveBuild(app, promise, cb) {
   if (typeof cb === 'function') {
-    promise.then(() => cb(null, app.taskState)).catch(cb);
+    promise.then(val => cb(null, val)).catch(cb);
     return;
   }
-  return promise.then(() => app.taskState);
```

Comment:
if no callback is passed, and the user does not return a value in the promise, will this still ensure that the state is returned? I didn't look at changes in the whole file yet.